components/dht/dht_memory.go

```diff
@@ -169,3 +169,24 @@ func (d *dht) replicaSlot(location1 SlotID) SlotID {
 	slotCount := len(d.slotVsNodes)
 	return SlotID((int(location1) + slotCount/2) % slotCount)
 }
+
+// ReassignMasterSlots marks all the slots owned by the failed node as followers and their corresponding replicas as leaders.
+// This method must be called only from the master.
+func (d *dht) ReassignMasterSlots(failedNode NodeID) map[SlotID]*SlotInfo {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	for slotID, slotInfo := range d.slotVsNodes {
```

Comment:
If we have multiple nodes replicating a slot, ain't we marking every one of them as Leader and what if the replica is already marked as unhealthy?